### ISSUE TITLE
JSON output contains syntax error when a filename in the target repo contains an escape character (Node)

### ISSUE THREAD

**Issue by jmensch1** [AlDanial/cloc#575]
## Overview

I ran `cloc` on https://github.com/systemd/systemd with the `--json` option. The repo contains [this file](https://github.com/systemd/systemd/blob/main/test/fuzz/fuzz-unit-file/dev-mapper-fedora_krowka%5Cx2dswap.swap) whose name includes an escape character. `cloc` ignored this file with the reason "language unknown". But due to the escape character in the filename, the `cloc` output file that lists ignored files could not be parsed to JSON.

I was able to reproduce this using the latest stable release (v1.88) on my mac.

## Steps to reproduce

```
# 1. clone the repo
> git clone https://github.com/systemd/systemd.git

# 2. run cloc with the json option 
> ./cloc --json --skip-uniqueness --by-file --report-file=cloc.json --ignored=ignored.json systemd
    4163 text files.
    4156 unique files.
Wrote ignored.json
    1390 files ignored.
Wrote cloc.json

# 3. enter Node repl and try to parse the ignored file 
> node 

>> data = fs.readFileSync('./ignored.json', 'utf-8')
[shows contents of ignored file]

>> JSON.parse(data) 
SyntaxError: Unexpected token x in JSON at position 48119
```

## A fix (at least for Node)

My text editor also couldn't parse the file as JSON, but it helped me track down the error to this part of the ignored file:
```
{"file":"systemd/test/fuzz/fuzz-unit-file/dev-mapper-fedora_krowka\x2dswap.swap", "reason":"language unknown (#3)"}
```
I found a fix for Node that involves replacing escape characters before parsing to JSON:
```
> node 
>> data = fs.readFileSync('./ignored.json', 'utf-8') 
>> JSON.parse(data.replace(/\\/g, '\\\\'))
[no error, shows parsed JSON]
```

**Comment by AlDanial** [comment]
(aside: if only all issues were so well written...)

This issue isn't straightforward at all since it involves locale settings on the git committer's end, how git handles utf-8 (or utf-16?) paths, and the cloc user's locale.  Regarding the file in question, it looks to me its author is Polish and the control character maps to a Polish character we don't see easily.  Just stripping it out seems wrong, yet I can't think of another way to produce correct JSON.  
I'll ponder this a bit and if nothing strikes me I suppose I'll have to strip non-printable ASCII characters.

**Comment by jmensch1** [comment]
Oh man, that sounds rough. Wish I had some insight. I googled character encoding the other night while falling asleep...and it helped me fall asleep.

By the way, in case you're curious about where your code ends up, I'm working on a Node/React app that visualizes code repositories. It relies on `cloc` pretty heavily. Here's a visualization of your repo:

![cloc](https://user-images.githubusercontent.com/9143823/111855231-d9057f00-88e0-11eb-8872-2a058cc91d09.png)

The code is [on github](https://github.com/jmensch1/codeflower) if you're interested.

Also worth noting that I've run `cloc` thousands of times (on a Node server) and this issue is the first time I got an error that had anything to do with `cloc`. Seems like a really narrow and technical issue. So thanks for some robust code :)
